src/utils.js

```diff
@@ -397,14 +398,29 @@ export function transition(el, stages) {
         requestAnimationFrame(() => {
             stages.end()
 
-            setTimeout(() => {
+            // Asign current transition to el in case we need to force it
+            el.__x_transition_remaining =() => {
+
                 stages.hide()
 
                 // Adding an "isConnected" check, in case the callback
                 // removed the element from the DOM.
                 if (el.isConnected) {
                     stages.cleanup()
                 }
+
+                // Safe to remove transition from el since it is completed
+                delete el.__x_transition_remaining
+                if(el.__x_transition_timer){
+                    clearTimeout(el.__x_transition_timer)
+                }
+            }
+
+            el.__x_transition_timer = setTimeout(() => {
+                // We only want to run remaining transitions in the end if they exists
+                if (el.__x_transition_remaining) {
+                    el.__x_transition_remaining()
+                }
```

Comment:
I'd really like to use the once function from VUE here. This feels a bit clumsy somehow. It's like we are trying to clear as many variables as possible to make sure the timeout doesn't run.
I'm almost sure we can get the same result with
```
el.__x_transition_remaining = once(() => {
    stages.hide()

    // Adding an "isConnected" check, in case the callback
    // removed the element from the DOM.
    if (el.isConnected) {
        stages.cleanup()
    }

     // Safe to remove transition from el since it is completed
     delete el.__x_transition_remaining
})

setTimeout(el.__x_transition_remaining, duration);
```

Reply:
Good news, i finally did figure out how to implement the way you mentioned. Good call @SimoTod. We are 1 variable less now :)